coriolis/tests/osmorphing/test_debian.py

```diff
@@ -0,0 +1,297 @@
+    def test_disable_predictable_nic_names_no_test_path(
+            self, mock_test_path, mock_read_file,
+            mock_write_file_sudo, mock_exec_cmd_chroot, mock_grub2_cfg_editor):
+
+        mock_test_path.return_value = False
+
+        self.morpher.disable_predictable_nic_names()
+
+        mock_read_file.assert_not_called()
+        mock_write_file_sudo.assert_not_called()
+        mock_exec_cmd_chroot.assert_not_called()
+        mock_grub2_cfg_editor.assert_not_called()
+
+    def test_get_update_grub2_command(self):
+        result = self.morpher.get_update_grub2_command()
+        self.assertEqual(result, "update-grub")
+
+    @mock.patch('coriolis.osmorphing.debian.StringIO')
+    def test__compose_interfaces_config(self, mock_stringio):
+        mock_stringio_instance = mock_stringio.return_value
+        result = self.morpher._compose_interfaces_config(self.nics_info)
+
+        mock_stringio_instance.write.assert_has_calls([
+            mock.call(debian.LO_NIC_TPL),
+            mock.call("\n\n"),
+            mock.call(debian.INTERFACES_NIC_TPL % {"device_name": "eth0"}),
+            mock.call("\n\n"),
+            mock.call(debian.INTERFACES_NIC_TPL % {"device_name": "eth1"}),
+            mock.call("\n\n")
+        ])
+        mock_stringio_instance.seek.assert_called_once_with(0)
+
+        self.assertEqual(result, mock_stringio_instance.read.return_value)
+
+    @mock.patch('yaml.dump')
+    def test__compose_netplan_cfg(self, mock_yaml_dump):
+        expected_cfg = {
+            "network": {
+                "version": 2,
+                "ethernets": {
+                    "lo": {
+                        "match": {
+                            "name": "lo"
+                        },
+                        "addresses": ["127.0.0.1/8"]
+                    },
+                    "eth0": {
+                        "dhcp4": True,
+                        "dhcp6": True,
+                    },
+                    "eth1": {
+                        "dhcp4": True,
+                        "dhcp6": True,
+                    }
+                }
+            }
+        }
+
+        result = self.morpher._compose_netplan_cfg(self.nics_info)
+
+        mock_yaml_dump.assert_called_once_with(
+            expected_cfg, default_flow_style=False)
+
+        self.assertEqual(result, mock_yaml_dump.return_value)
+
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_write_file_sudo')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_exec_cmd_chroot')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_test_path')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_compose_netplan_cfg')
+    @mock.patch.object(debian.BaseDebianMorphingTools, '_list_dir')
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, 'disable_predictable_nic_names'
+    )
+    @mock.patch.object(
+        debian.BaseDebianMorphingTools, '_compose_interfaces_config'
+    )
+    def test_set_net_config(
```

Comment:
This test case can be written without mocking `_compose_netplan_cfg` and `_compose_interfaces_config`, but you have to adjust the inputs and assertions.